### ISSUE TITLE
html() from kwic output doesn't work if no template is provided

### ISSUE THREAD

**Issue by ChristophLeonhardt** [PolMine/polmineR#126]
### tldr

I think that `read()` for `kwic` objects right now is highly dependent on a working template file and in the end I am not sure if that is really an issue or just an inconvenience for people who forgot to add one when creating a corpus. 

---

### Long version:

In the UCSSR slides, it is mentioned that it is possible to get the full text from a `kwic` object by using the `read()` method, such as:

```
bt2005 <- partition("GERMAPARL", year = 2005)
K <- kwic(bt2005, query = "Ausländer")
read(K, i = 1) 
```

This, indeed, works as long you use a corpus which does have a template defining how the `read()` output is formatted. It does not work if no template is provided and no metadata is passed when creating the `kwic` object.

This seems to be because the underlying `html()` function is dependent on one of the following three things:
* an explicitly stated s_attribute (which is not passed by `read()`)
* metadata in the metadata slot of the `kwic` object
* s-attributes defined by the template in `    s_attrs <- getOption("polmineR.templates")[[object@corpus]][["metadata"]]`

Working with a corpus which does not have a template, this gets a bit buggy.

```
unga_2005 <- partition("UNGA", year = 2005)
K <- kwic(unga_2005, query = "immigration")
read(K, i = 1)
```

>  Error in (function (cl, name, valueClass)  : 
  assignment of an object of class “NULL” is not valid for @‘s_attribute_strucs’ in an object of class “plpr_partition”; is(value, "character") is not TRUE 

Which is thrown here in the `html` method:

```
  partition_to_read <- partition(
    object@corpus,
    def = lapply(setNames(s_attrs, s_attrs), function(x) object@stat[[x]][i]),
    type = type
  )
```

because here, the s_attrs object is empty.

If you add an s-attribute when creating the `kwic` object, another error occurs:

```
unga_2005 <- partition("UNGA", year = 2005)
K <- kwic(unga_2005, query = "immigration", s_attribute = "speaker")
read(K, i = 1)
```

>  Error in check_s_attribute(corpus = corpus, registry = registry, s_attribute = s_attribute) : 
  s_attribute needs to be a length 1 vector 

This happens somewhere in the `as.markdown()` portion of the subsequent `html()` method for partition objects, but here I have to admit that I am guessing when I say that this is related to this line in the `as.markdown()` method for the `plpr_subcorpus` class which again checks for the template and determines that the s_attribute is NULL. 

```
  type <- cl_struc2str(corpus = .Object@corpus, s_attribute = template[["speech"]][["sAttribute"]], struc = .Object@strucs, registry = registry())
```

Maybe you quickly see how these things are related. Unfortunately, I am not entirely sure what to make of it.

**Comment by PolMine** [comment]
The templates for the fulltext output are a nasty and somewhat opaque part of the package, I am afraid. To start making improvements, I dropped a workaround that would assign a elementary template if no template is available.  So in the latest dev version, you will see whether a template is available, or not.

```r
library(polmineR)
use("UNGA")
corpus()
```

Then I introduced telling warnings, if no template is found for a corpus.

```r
unga_2005 <- partition("UNGA", year = 2005)
K <- kwic(unga_2005, query = "immigration", meta = c("speaker", "date"))
read(K, i = 1)
```

But of course you may want to set a template. Using the template of the GERMAPARL corpus for UNGA would work like this.

```r
options(
  polmineR.templates = c(
    getOption("polmineR.templates"),
    list("UNGA" = getOption("polmineR.templates")[["GERMAPARL"]]))
)
read(K, i = 1)
```

But this is not yet a proper solution. The documentation of the relevant methods (`get_template()` and `set_template()`) needs to be improved, and presumably, there needs to be a vignette that explains in depth how you could define a template for your corpus.

**Comment by ChristophLeonhardt** [comment]
Testing this on `polmineR 0.8.0.9016`, I am not sure if I can reproduce your success?

This workflow: 

```
library(polmineR)
use("UNGA")
# use("GermaParl")
corpus()

unga_2005 <- partition("UNGA", year = 2005)
K <- kwic(unga_2005, query = "immigration", meta = c("speaker", "date"))

options(
  polmineR.templates = c(
    getOption("polmineR.templates"),
    list("UNGA" = getOption("polmineR.templates")[["GERMAPARL"]]))
)
read(K, i = 1)
```

still results in the error:

>   s_attribute needs to be a length 1 vector 

I also cannot provoke a warning about the missing template.

**Comment by ablaette** [comment]
I just checked whether this works:

```r
unga_2005 <- partition("UNGA", year = 2005)
K <- kwic(unga_2005, query = "immigration", meta = c("speaker", "date"))
read(K, i = 1)
```

And it does. For a `kwic` object without s-attributes, you get an expressive warning.
